crates/router/src/core/errors/utils.rs

```diff
@@ -21,6 +21,7 @@ impl StorageErrorExt for error_stack::Report<errors::StorageError> {
             self.change_context(not_found_response)
         } else {
             self.change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Unable to resolve as database not found")
```

Comment:
This block basically means if the database returned a `NotFound` error, then change the context to be a `PaymentNotFound`, `MerchantAccountNotFound`, etc. response. In other cases, return an `InternalServerError`. 

This is not about failing to resolve as "database not found".